Providers/Profile.swift

```diff
@@ -1018,9 +1064,19 @@ open class BrowserProfile: Profile {
             public var description = "Failed to get stored key."
         }
 
+        public class DeviceIdError: MaybeErrorType {
+            public var description = "Failed to get deviceId."
+        }
+
         fileprivate func syncUnlockInfo() -> Deferred<Maybe<SyncUnlockInfo>> {
             let d = Deferred<Maybe<SyncUnlockInfo>>()
             profile.rustFxA.accountManager.uponQueue(.main) { accountManager in
+                guard let deviceId = accountManager.deviceConstellation()?.state()?.localDevice?.id else {
+                    SentryIntegration.shared.sendWithStacktrace(message: "Device Id could not be retrieved", tag: SentryTag.rustRemoteTabs, severity: .warning)
+                    d.fill(Maybe(failure: DeviceIdError()))
```

Comment:
I know you didn't create it, but can we rename this to something more legible? Single letter variable names are not informative.